models/src/main/scala/coop/rchain/models/NormalizerEnv.scala

```diff
@@ -0,0 +1,25 @@
+package coop.rchain.models
+
+import com.google.protobuf.ByteString
+import coop.rchain.casper.protocol.DeployDataProto
+import coop.rchain.crypto.PublicKey
+import coop.rchain.models.rholang.implicits._
+
+object NormalizerEnv {
+  type UriString     = String
+  type NormalizerEnv = Map[UriString, Par]
```

Comment:
Now, I actually have an objection. I am not sure if this is the right type to model the environment. Why?
1. there is no control over what's in it
1. I have to always check if I can get the value I **know** I put in (or must be in).

I am not sure if this is overkill but can we possibly model it like shapeless record (with literal Strings as keys) and have the precise type of the Env for System Deploys?
I am taking a `no` for an answer BTW :smile: @JosDenmark @ArturGajowy @tymm ?

Reply:
It could actually be `Map[UriString, Expr | GUnforgeable]`.
However it would make the Protobuf and possibly the `HashM` and `EqualM` instances more complex.

If no one has any objections, I would prefer creating a ticket for that and come back to that in the future.

Reply:
@marcin-rzeznicki I relate, but this PR is blocking the rest of the cost-accounting work.

Reply:
I hear you gents - rest of it is looking great. Approved